frontend/src/components/Page.test.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+  This file is part of Edgehog.
+
+  Copyright 2023 SECO Mind Srl
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  SPDX-License-Identifier: Apache-2.0
+*/
+
+import { describe, it, expect, vi } from "vitest";
+import { screen } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import { renderWithProviders } from "setupTests";
+
+import Page from "./Page";
+
+describe("Header", () => {
+  it("renders title correctly", () => {
+    renderWithProviders(<Page.Header title="Header Title" />);
+
+    const pageTitle = screen.getByRole("heading", { name: "Header Title" });
+    expect(pageTitle).toBeVisible();
+    expect(pageTitle).toHaveTextContent("Header Title");
+  });
+
+  it("does not render title if title not given", () => {
+    renderWithProviders(<Page.Header />);
+
+    const pageTitle = screen.queryByRole("heading", { name: "" });
+    expect(pageTitle).not.toBeInTheDocument();
+  });
+
+  it("renders children correctly", () => {
+    const childText = "Test Children";
+    renderWithProviders(<Page.Header>{childText}</Page.Header>);
+
+    const childElement = screen.getByText(childText);
+    expect(childElement).toBeVisible();
+  });
+});
+
+describe("Main", () => {
+  it("renders children correctly", () => {
+    const childText = "Test Children";
+    renderWithProviders(<Page.Main children={childText} />);
+
+    const childElement = screen.getByRole("main");
+    expect(childElement).toBeVisible();
+  });
```

Comment:
It does not test whether the children is rendered. We can check the children as in the test for header above. We can also check for the `main` role in a separate test.
```suggestion
  it("indicates the primary content of a document", () => {
    renderWithProviders(<Page.Main />);

    expect(screen.getByRole("main")).toBeVisible();
  });

  it("renders children correctly", () => {
    renderWithProviders(
      <Page.Main>
        <div data-testid="main-children" />
      </Page.Main>,
    );

    expect(screen.getByTestId("main-children")).toBeVisible();
  });
```